azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_ops_provisioning.py

```diff
@@ -33,15 +33,40 @@ def __init__(self, security_client, callback=None):
 # TODO : Should this be named SetSecurityProviderArgs or SetSecurityClientArgs?
 
 
+class SetX509SecurityClient(PipelineOperation):
```

Comment:
Okay, I'm a little confused about some stuff here

`SetX509SecurityClient` gets turned into `SetSecurityClientArgs` in `pipeline_stages_provisioning`, so it seems these are kind of redundant.

I could understand if `SetSecurityClientArgs` was a generic operation that applied to all security stuff, but despite the docs saying it applies also to `SetSymmetricKeySecurityClient`, this does not actually seem to be the case, as instead it uses the specific `SetSymmetricKeySecurityClientArgs`.

Reply:
sorry for so much confusion. I should have described in the PR that I wanted opinion on that TODO. I hope using generic ``SetSecurityClientArgs`  will resolve the confusion.

---
In reply to: [291748870](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291748870) [](ancestors = 291748870)